accio-sqlrewrite/src/main/java/io/accio/sqlrewrite/AccioDataLineage.java

```diff
@@ -246,14 +266,18 @@ protected Void visitDereferenceExpression(DereferenceExpression node, Void ignor
                     // collect join keys
                     for (int i = 0; i < info.getRelationships().size(); i++) {
                         Relationship relationship = info.getRelationships().get(i);
-                        String modelName = relationship.getModels().get(1);
-                        String joinKey = getJoinKey(parseExpression(relationship.getCondition()), modelName).orElseThrow();
-                        sourceColumns.put(modelName, joinKey);
+                        String left = relationship.getModels().get(0);
+                        String right = relationship.getModels().get(1);
+                        String leftKey = getJoinKey(parseExpression(relationship.getCondition()), left).orElseThrow();
+                        String rightKey = getJoinKey(parseExpression(relationship.getCondition()), right).orElseThrow();
```

Comment:
We can parse it just once.
```suggestion
                        Expression joinCondition = parseExpression(relationship.getCondition());
                        String leftKey = getJoinKey(joinCondition, left).orElseThrow();
                        String rightKey = getJoinKey(joinCondition, right).orElseThrow();
```